pkg/helmraiser/helm.go

```diff
@@ -48,114 +69,98 @@ func confToArgs(conf HelmConf) ([]string, []string, error) {
 	// append custom flags to args
 	args = append(args, conf.Flags...)
 
-	if len(args) == 0 {
-		args = nil
-	}
-
 	return args, tempFiles, nil
 }
 
-func parseYamlToMap(yamlFile []byte) (map[string]interface{}, error) {
-	files := make(map[string]interface{})
-	d := yaml.NewDecoder(bytes.NewReader(yamlFile))
-	for {
-		var doc, jsonDoc interface{}
-		if err := d.Decode(&doc); err != nil {
-			if err == io.EOF {
-				break
-			}
-			return nil, errors.Wrap(err, "parsing manifests")
-		}
-
-		jsonRaw, err := json.Marshal(doc)
-		if err != nil {
-			return nil, errors.Wrap(err, "marshaling mainfests")
-		}
+// Template expands a Helm Chart into a regular manifest.List using the `helm
+// template` command
+func (h Helm) Template(name, chart string, opts TemplateOpts) (manifest.List, error) {
+	confArgs, tmpFiles, err := confToArgs(opts)
+	if err != nil {
+		return nil, err
+	}
+	for _, f := range tmpFiles {
+		defer os.Remove(f)
+	}
 
-		if err := json.Unmarshal(jsonRaw, &jsonDoc); err != nil {
-			return nil, errors.Wrap(err, "unmarshaling manifests")
-		}
+	args := []string{name, chart}
+	args = append(args, confArgs...)
 
-		// Unmarshal name and kind
-		kindName := struct {
-			Kind     string `json:"kind"`
-			Metadata struct {
-				Name string `json:"name"`
-			} `json:"metadata"`
-		}{}
-		if err := json.Unmarshal(jsonRaw, &kindName); err != nil {
-			return nil, errors.Wrap(err, "subtracting kind/name through unmarshaling")
-		}
+	cmd := h.run("template", args...)
+	var buf bytes.Buffer
+	cmd.Stdout = &buf
+	cmd.Stderr = os.Stderr
 
-		// snake_case string
-		normalizeName := func(s string) string {
-			s = strings.ReplaceAll(s, "-", "_")
-			s = strings.ReplaceAll(s, ":", "_")
-			s = strings.ToLower(s)
-			return s
-		}
+	if err := cmd.Run(); err != nil {
+		return nil, errors.Wrap(err, "Expanding Helm Chart")
+	}
```

Comment:
This could all be wrapped in the Helm.run() function, I would even think of returning `func (c *Cmd) Output() ([]byte, error)` for this.

Reply:
(bit awkward that github diff included the normalizeName, just ignore that)

Reply:
Reusing the same pattern that proved working well in `pkg/kubernetes/client`: https://github.com/grafana/tanka/blob/3ff6c70fae6252d9d0e391b3831474cba2e04336/pkg/kubernetes/client/exec.go#L21

Returning the raw `exec.Cmd` allows consuming functions to also modify the commands environment, stdin, capture stderr if required, etc.

I think we will need this in the future here.